bigframes/session/bq_caching_executor.py

```diff
@@ -572,6 +602,21 @@ def _execute_plan(
         )
 
 
+def _if_schama_match(
+    table_schema: Tuple[bigquery.SchemaField, ...], schema: schemata.ArraySchema
+) -> bool:
+    if len(table_schema) != len(schema.items):
+        return False
+    for field in table_schema:
+        if field.name not in schema.names:
+            return False
+        if bigframes.dtypes.convert_schema_field(field)[1] != schema.get_type(
```

Comment:
Do we need to do anything special here for the duration/timedelta type that we added recently?